### PR TITLE
Flypeople are now FOV immune

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Eyes now give FOV out, and Flypeople eyes are immune to it.

## How This Contributes To The Skyrat Roleplay Experience

Flies generally can see all around them with a near 360 point of view, so I thought we can maybe reflect that in-game as well, making them more unique as a species.
Flypeople also can't see very well, which is why they are flashable by all directions, something I added in https://github.com/tgstation/tgstation/pull/55724

## Changelog

:cl:
balance: Fly eyes don't have an innate field of view anymore.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/living_fov.dm [hunk 1]
```diff
@@ -45,14 +45,14 @@
...
+	if(. && is_nearsighted())
 		if((rel_x >= NEARSIGHTNESS_FOV_BLINDNESS || rel_x <= -NEARSIGHTNESS_FOV_BLINDNESS) || (rel_y >= NEARSIGHTNESS_FOV_BLINDNESS || rel_y <= -NEARSIGHTNESS_FOV_BLINDNESS))
 			return FALSE
 
 /// Updates the applied FOV value and applies the handler to client if able
 /mob/living/proc/update_fov()
 	var/highest_fov
-	if(CONFIG_GET(flag/native_fov))
+	if(CONFIG_GET(flag/native_fov) && !HAS_TRAIT(src, FOV_IMMUNE))
 		highest_fov = native_fov
```
Comment (Zonespace27) [inline code/modules/mob/living/living_fov.dm:56]: skyrat edit comment

File: code/modules/surgery/organs/eyes.dm [hunk 2]
```diff
@@ -37,13 +37,19 @@
 	/// indication that the eyes are undergoing some negative effect
 	var/damaged = FALSE
 
+	///Whether the pair of eyes can see in all directions.
+	var/see_all_dirs = FALSE //SKYRAT EDIT - FOV immune eyes
```
Comment (Zonespace27) [inline code/modules/surgery/organs/eyes.dm]: do the comment like
```
// SKYRAT EDIT START
stuff
stuff2
// SKYRAT EDIT END
```
when doing multi-line things, even comments

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
Based willard PR

**Comment by GizkaFreechman** [comment]
this is a pretty significant buff, especially when almost all fly person race traits are negative and im not saying this because im a fly main

-they take desword damage from fly swatters (guaranteed hairline fracture, after one hit) 
-pest spray and the bug spray drink, deal significant toxin damage
-their livers cause them to get drunk easier
-they have slower movespeed? (needs citation)
-organ names are randomized
-the fly stomach is the culprit behind all the puking
-the Proboscis tongue organ, is what they use to slurp that vomit
-their eyes more vulnerable to flashes (as stated in the pr).

all this, just for better FOV, I'm gonna take what I can get

**Comment by ghost** [comment]
Couldnt this be done on TG as innate fov is a config anyways?

**Comment by JohnFulpWillard** [comment]
> Couldnt this be done on TG as innate fov is a config anyways?

Tbh I didn't even realize this was a config until I started working on it, I'll try it now.

**Comment by hugmeorelse** [comment]
I'm fine with this, considering nobody really plays flypersons all too much, and it makes sense for them. However, I'm not sure about full 360 degree FOV.

**Pull request closed without merging**